notebooks/02-ingestion/04-table-datatypes.ipynb

```diff
@@ -0,0 +1,851 @@
+    "SELECT\n",
+    "  TIME_PARSE(\"depaturetime\") AS \"__time\",\n",
+    "  \"Year\",\n",
+    "  \"Reporting_Airline\",\n",
+    "  \"Origin\",\n",
+    "  \"Dest\",\n",
+    "  \"Distance\"\n",
+    "FROM \"ext\"\n",
+    "WHERE \"depaturetime\" <> 0\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "\n",
+    "display.run_task(sql)\n",
+    "sql_client.wait_until_ready('example-flights-types-1')\n",
+    "display.table('example-flights-types-1')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "889be0da-36ec-4510-876a-c7c040355b37",
+   "metadata": {},
+   "source": [
+    "You can query the `INFORMATION_SCHEMA` schema's `COLUMNS` table to [get information](https://druid.apache.org/docs/latest/querying/sql-metadata-tables#columns-table) for any TABLE in Druid.\n",
+    "\n",
+    "Run this cell which queries this table to return further information about the underlying structure of the `example-flights-types-1` table."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3a375431-92a0-42d3-901f-434060e9ff57",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  \"COLUMN_NAME\",\n",
+    "  \"ORDINAL_POSITION\",\n",
+    "  \"DATA_TYPE\",\n",
+    "  \"NUMERIC_PRECISION\",\n",
+    "  \"NUMERIC_PRECISION_RADIX\",\n",
+    "  \"DATETIME_PRECISION\",\n",
+    "  \"CHARACTER_SET_NAME\",\n",
+    "  \"JDBC_TYPE\"\n",
+    "FROM \"INFORMATION_SCHEMA\".\"COLUMNS\"\n",
+    "WHERE \"TABLE_NAME\" = 'example-flights-types-1'\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "db0b419f-8de5-4ee4-8b2a-fcf4551af932",
+   "metadata": {},
+   "source": [
+    "The type shown in the `DATA_TYPE` column tells you how Druid will interpret the data in SQL.\n",
+    "\n",
+    "As shown in [documentation](https://druid.apache.org/docs/latest/querying/sql-data-types), these SQL types map to more fundamental types inside Druid itself. Before going further in this notebook, take a moment to use the [documentation](https://druid.apache.org/docs/latest/querying/sql-data-types#standard-types) to understand how each `DATA_TYPE` in the `example-flights-types-1` TABLE maps to an internal Druid runtime type.\n",
+    "\n",
+    "The rest of this notebook works through examples of these data types:\n",
+    "\n",
+    "* [Numbers](#numbers) (e.g. TIMESTAMP, BOOLEAN, DOUBLE, FLOAT, LONG)\n",
+    "* [Strings](#strings) (e.g. STRING)\n",
+    "* [Arrays](#arrays) (ARRAY)\n",
+    "* [Other](#other) (COMPLEX)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "018e3239-7276-4236-bf6e-b63194ad5957",
+   "metadata": {},
+   "source": [
+    "## Numbers\n",
```

Comment:
I think it would be good to explain why "isSilicon" is stored as a "BIGINT" and not "BOOLEAN".. Druid does not have a Boolean data type for its internal representation so it stores it as a BIGINT with two values, 0 and 1.

Reply:
Disregard, I see that you explain it further below....